persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/connectionfactory/PooledConnectionFactory.java

```diff
@@ -47,15 +43,38 @@ public void start(ConnectionFactoryConfiguration config, ClassLoader classLoader
                                               "PooledConnectionFactoryConfiguration");
       }
 
-      connectionPool = C3P0ConnectionPool.forceC3P0() ? new C3P0ConnectionPool(classLoader, poolConfig) :
-            new HikariConnectionPool(classLoader, poolConfig);
-      if (trace) log.tracef("Started connection factory with config: %s", config);
+      try {
+
+         String propsFile = poolConfig.propertyFile();
+         if (propsFile != null) {
+            dataSource = AgroalDataSource.from(new AgroalPropertiesReader(PROPERTIES_PREFIX).readProperties(propsFile));
+         } else {
+            // Default Agroal configuration with a fixed size connection-pool and metrics disabled
+            String password = poolConfig.password() != null ? poolConfig.password() : "";
+            AgroalDataSourceConfigurationSupplier configuration = new AgroalDataSourceConfigurationSupplier()
+                  .connectionPoolConfiguration(cp -> cp
+                        .minSize(10) // Same as maxSize so we have a fixed size connection pool to maximize responsiveness
```

Comment:
I think this is unnecessarily aggressive. If the user wants tuning they can use the props file.

Reply:
Yeah fair enough, I was just re-using the HikariCP default settings there. Updated.